### PR TITLE
Add electrodes interface for Huszar

### PR DESCRIPTION
**Pull request by h-mayorquin**
This will add electrode information even when the Neuroscope interface for the raw data is no present.

### CODE REVIEW

File: buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/electordesinterface.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        recording = NumpyRecording(
+            traces_list=traces_list, sampling_frequency=sampling_frequency, channel_ids=channel_ids
+        )
+        recording.set_channel_locations(channel_ids=channel_ids, locations=locations)
+
+        recording.set_property(key="channel_name", values=channel_name)
+
+        # This should transform to microvolts which is the standard in spikeinterface
+        recording.set_channel_gains(channel_ids=channel_ids, gains=np.ones(num_channels) * 0.195)
+        recording.set_channel_offsets(channel_ids=channel_ids, offsets=np.ones(num_channels) * 32_768)
```
Comment (CodyCBakerPhD) [inline buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/electordesinterface.py:51]: Whoa whoa, do offsets get set to the real extractor? Or to a corresponding intan reading of the info.rhd?

Thing is, this is signed data - usually that offset is only for unsigned intan data

I could have sworn that when using SI to read this stuff the offset is only set if its unsigned

File: buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/electordesinterface.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        recording.set_property(key="group", ids=channel_ids, values=channel_groups)
+        recording.set_property(key="group_name", ids=channel_ids, values=channel_group_names)
+
+        add_electrodes(recording=recording, nwbfile=nwbfile)
+
+        x = 1
```
Comment (CodyCBakerPhD) [inline buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/electordesinterface.py:57]: ?

Comment (h-mayorquin) [reply buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/electordesinterface.py:57]: Wrong, left for debugging.

File: buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/electordesinterface.py (hunk 1 above)
Comment (h-mayorquin) [reply buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/electordesinterface.py:51]: Thanks for pointing that out. I am losing a multiplicative factor. Look at how the oficial implementation for reading intan that intan systems provides looks:

https://github.com/Intan-Technologies/load-rhd-notebook-python/blob/9026c8c0ac0693ed4896c2f50f55a3f57237bb68/importrhdutilities.py#L579

But I think they use the offset there is because they initially define the data as unsigned:
https://github.com/Intan-Technologies/load-rhd-notebook-python/blob/9026c8c0ac0693ed4896c2f50f55a3f57237bb68/importrhdutilities.py#L511

Which sounds like what are you saying as Neuroscope already loads the data signed and then has the offset 0.
https://github.com/NeuralEnsemble/python-neo/blob/f608309c5ce031ecd905349c140b07a3dafb057d/neo/rawio/neuroscoperawio.py#L111

I have become very suspicious of that Neuroscope reader but if indeed we are going to read the data using it, we should probably set the offest to 0.

The other readers for intan have the offset though:

pyintan has it though:
https://github.com/alejoe91/pyintan/blob/92ee10acad13bd2b5207a158a2e38ce222be716a/pyintan/intan.py#LL108C19-L108C19

And the reader for intan in neo:
https://github.com/NeuralEnsemble/python-neo/blob/f608309c5ce031ecd905349c140b07a3dafb057d/neo/rawio/intanrawio.py#L491

File: buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/electordesinterface.py (hunk 2 above)
Comment (h-mayorquin) [reply buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/electordesinterface.py]: Removed.

File: buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/electordesinterface.py (hunk 1 above)
Comment (CodyCBakerPhD) [reply buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/electordesinterface.py]: I know the base readers underlying pyintan and neo always set it, my contention was that something in the SI loader would make the decision whether or not to set that to the extractor, somewhere around https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/extractors/neoextractors/neobaseextractor.py#L156

At least the last time I used the intan reader on data, that's how it worked (I've ran before on both signed and unsigned intan data) - point being, the offset was only needed when the data was unsigned

Anyway, yes for Neuroscope it should soundly be zero, since it was acquired as signed intan (which also should not use offsets, which if SI currently does is, I suspect, a bug)

Comment (h-mayorquin) [reply buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/electordesinterface.py]: I commented out the line because we are going to use Neuroscope. I also tested Neuroscope for the valero data and the offset is zero.

Beside that, 
Are you saying that using SI to read intan data should set the offsets to zero? Not clear to me why and it is not the case if I read Valero data with Intan:

image

I don't really understand what do you mean by "acquiring with signed intan", can you elaborate?

Comment (h-mayorquin) [reply buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/electordesinterface.py]: Got you, yeah, I am confused about this. Pyintan has this which indicates that there are some versions where the data is typed:

https://github.com/NeuralEnsemble/python-neo/blob/f608309c5ce031ecd905349c140b07a3dafb057d/neo/rawio/intanrawio.py#L473-L491

But I can't find any references to it in the oficial repo provided by the company...

**Comment by garrettmflynn** [comment]
@CodyCBakerPhD Uploaded a stub file with preliminary ecephys metadata to DANDI for this project: https://dandiarchive.org/dandiset/000552?pos=1

**Comment by h-mayorquin** [comment]
> @CodyCBakerPhD Uploaded a stub file with preliminary ecephys metadata to DANDI for this project: https://dandiarchive.org/dandiset/000552?pos=1

Did this work?

**Comment by CodyCBakerPhD** [comment]
> Did this work?

I'm having him try not using the ElectrodeInterface and use the new CellExplorerSortingInterface from NeuroConv to see what that looks like

Even after fixing the group indexing issue, Garretts latest file still had that channel name issue from before
